src/server/game/AI/SmartScripts/SmartScriptMgr.h

```diff
@@ -1479,15 +1473,13 @@ class TC_GAME_API SmartWaypointMgr
 
         void LoadFromDB();
 
-        WPPath* GetPath(uint32 id)
+        WPPath const &GetPath(uint32 id)
         {
-            if (waypoint_map.find(id) != waypoint_map.end())
-                return waypoint_map[id];
-            else return nullptr;
+            return waypoint_map[id];
```

Comment:
it's gonna invalidate iterators tho if it triggers a re-hash. it also modifies a shared container in a thread-unsafe environment (it seems GetPath() can be called in Map::Update())